docs/source/api/apollo-server.md

```diff
@@ -343,15 +343,44 @@ addMockFunctionsToSchema({
    to standard error. Specify this function to process errors in a different
    way.
 
-*  `privateVariables`: Array<String> | boolean
+*  `privateVariables`: Array<String\> | boolean
 
+   DEPRECATING IN VERSION XX.XX.XX for `sendVariableValues`, which will support the same
+   functionalities but allow for more flexibility.
+   
    A case-sensitive list of names of variables whose values should not be sent
    to Apollo servers, or 'true' to leave out all variables. In the former
    case, the report will indicate that each private variable was redacted in
    the latter case, no variables are sent at all.
+   
+* `sendVariableValues`: { valueModifier: (options: { variables: Record<string, any>, operationString?: string } ) => Record<string, any> }
+                     | { exceptVariableNames: Array<String\> }
+                     | { safelistAll: boolean }
+    
+    By default, Apollo Server does not send the values of any GraphQL variables to Apollo's servers, because variable values often contain the private data of your app's users. If you'd like variable values to be included in traces, set this option. This option can take several forms:
+    
+    - { safelistAll: ... }: false to blocklist, or true to safelist all variable values
+    - { valueModifier: ... }: a custom function for modifying variable values
+    - { exceptVariableNames: ... }: a case-sensitive list of names of variables whose values should not be sent to Apollo servers
+
+    Defaults to blocklisting all variable values if both this parameter and
+    the to-be-deprecated `privateVariables` are not set. The report will also
+    indicate each private variable key redacted by { safelistAll: false } or { exceptVariableNames: [...] }.
+    
+* `sendHeaders`: { except: Array<String\> } | { safelistAll: boolean }
+   By default, Apollo Server does not send the list of HTTP headers and values to
+   Apollo's servers, to protect private data of your app's users. If you'd like this information included in traces,
+   set this option. This option can take two forms:
+   
+   - {except: Array<String\>} A case-insensitive list of names of HTTP headers whose values should not be
+   sent to Apollo servers
+   - {safelistAll: true/false} to include or leave out all HTTP headers.
+  
+   Unlike with sendVariableValues, names of dropped headers are not reported.
```

Comment:
We should document the existing behavior where `authorization`, `cookie`, and `set-cookie` are never sent.